devices/bosch/room_thermostat2.json

```diff
@@ -90,12 +90,42 @@
             "fn": "zcl:attr"
           }
         },
+        {
+          "name": "config/coolsetpoint",
+          "refresh.interval": 360
+        },
         {
           "name": "config/checkin",
           "public": false,
           "refresh.interval": 3600,
           "default": 14400
         },
+        {
+          "name": "config/devicemode",
+          "refresh.interval": 3600,
+          "read": {
+            "at": "0x001C",
+            "cl": "0x0201",
+            "ep": 1,
+            "fn": "zcl:attr"
+          },
+          "write": {
+            "at": "0x001C",
+            "cl": "0x0201",
+            "dt": "0x30",
+            "ep": 1,
+            "eval": "if (Item.val == 'off') { 0 } else if (Item.val == 'cool') { 3 } else if (Item.val == 'heat') { 4 };",
+            "fn": "zcl:attr"
+          },
+          "parse": {
+            "at": "0x001C",
+            "cl": "0x0201",
+            "ep": 1,
+            "eval": "if (Attr.val == 0) { Item.val = 'off' } else if (Attr.val == 3) { Item.val = 'cool' } else if (Attr.val == 4) { Item.val = 'heat' };",
+            "fn": "zcl:attr"
+          },
+          "default": "heat"
+        },
```

Comment:
Is this confirmed to be working? Bosch already seems to use a manufacturer specific attribute to set the heating modes and apparently disregards the standard attribute.

Reply:
At least it works for me. The two channels "heat/cool" then also use the normal modes: auto, manual(heat) and off.


The attribute “devicemode” was also obvious to me.

Reply:
But what do I do here with heat and the off(450) problem?

Reply:
How about if I use mode/preset instead of devicemode/mode?

mode --> heat, cool and off
preset --> manual, auto and holiday